### ISSUE TITLE
wudsn-ide-install script failing on OSX

### ISSUE THREAD

**Issue by tonual** [wudsn/wudsn-ide-install#2]
Hello, 
thank you for enormous effort of bringing this toolkit to the 8bit era fans.

I am on Apple M2 silicon, Sonoma 14.4 OS and already tried running wudsn-ide-install.
Because it failed at some point, I haven't delved into troubleshooting, but rather tried other way around
using manual installation described at https://www.wudsn.com/index.php/ide/installation. 

It was kind of OK, but once I've discovered some features where not working as advertised  (e.g content assistance)
I decided to create an issue for wudsn-ide-install script hoping there is a simple fix and try again.

For what I could observed the script executes just fine until the window pops up with an 'Eclipse' and 'Application' folder icons.
(probably wanting my to perform drag&drop) but in the meantime the script fails immediately.

Here is a fragment from terminal from start till it failed:

`
Installing WUDSN IDE feature.
/Users/rafalkrzysztof/retro_dev/WUDSN/wudsn.command: line 377: 21192 Killed: 9               ${ECLIPSE_EXECUTABLE} -nosplash -application org.eclipse.equinox.p2.director -repository "${UPDATE_URL}" -installIU com.wudsn.ide.feature.feature.group -destination "$ECLIPSE_DESTINATION_FOLDER" >> "${LOG}" 2>> "${LOG}"
ERROR: See messages above and in /Users/rafalkrzysztof/retro_dev/WUDSN/wudsn.log.
`

Attaching log file. 

[wudsn.log](https://github.com/wudsn/wudsn-ide-install/files/15112032/wudsn.log)

**Comment by peterdell** [comment]
Thanks a lot for your feedback. I have a macOS M1 which I'll update and try to reproduce the problem.

**Comment by peterdell** [comment]
Hmm, I get these warning some lines before the error message on the console
Installing Eclipse.

WARNING: The locate database (/var/db/locate.database) does not exist.
To create the database, run the following command:

  sudo launchctl load -w /System/Library/LaunchDaemons/com.apple.locate.plist

Please be aware that the database can take some time to generate; once
the database has been created, this message will no longer appear.

Installing WUDSN IDE feature.
/Users/jac/jac/wudsn/daily/wudsn.command: line 377:  2608 Killed: 9               ${ECLIPSE_EXECUTABLE} -nosplash -application org.eclipse.equinox.p2.director -repository "${UPDATE_URL}" -installIU com.wudsn.ide.feature.feature.group -destination "$ECLIPSE_DESTINATION_FOLDER" >> "${LOG}" 2>> "${LOG}"
ERROR: See messages above and in /Users/jac/jac/wudsn/daily/wudsn.log.
W

**Comment by peterdell** [comment]
Looks like the OS is killing the "eclipse" process because of code signing. Not sure why, because removing the signature should avoid it.
    # Check is codesign is completely installed.
    if command -v "codesign_allocate" &> /dev/null; then
      # Remove code signing information to prevent issues when changing files in the folder.
      codesign --remove "${ECLIPSE_APP_FOLDER}"
    fi

JACs-Mac-mini:MacOS jac$ codesign -d eclipse 
eclipse: code object is not signed at all

https://support.apple.com/en-us/102445
https://bugs.eclipse.org/bugs/show_bug.cgi?id=494293
https://github.com/espressif/idf-eclipse-plugin/issues/780
https://forums.developer.apple.com/forums/thread/684509

Here's a workaround that codesigns the Eclipse.app folder, but that can break at any time as Eclipse writes files to it.

[wudsn.command.zip](https://github.com/wudsn/wudsn-ide-install/files/15180773/wudsn.command.zip)

**Comment by tonual** [comment]
Thank you for investigating the issue. 
I understand Apple ecosystem moves too fast to catch up with compliance on the low level scripting.

I decided to try manual installation and it was kind of successful; I have a syntex coloring, nice Outline window with code navigation, super fast compiler and code execution with emulator. What more could I want? :)

The only feature not working correctly is code hinting/completition.
It seems to start hinting only after typing the dot as first character, but it does not e.g hint the (declared) equates. What could be missing there ?

![scr1_](https://github.com/wudsn/wudsn-ide-install/assets/7768922/8dc6dccb-e4bc-44ff-ae39-690652b17412)
 
![scr2](https://github.com/wudsn/wudsn-ide-install/assets/7768922/92b534c2-d676-4632-8439-8f307101b44b)

<img width="936" alt="Screenshot 2024-05-03 at 10 15 27" src="https://github.com/wudsn/wudsn-ide-install/assets/7768922/fdda5aee-2584-44c3-969d-ee798a61c714">

<img width="921" alt="Screenshot 2024-05-03 at 10 16 47" src="https://github.com/wudsn/wudsn-ide-install/assets/7768922/2d372cd0-56d9-45fe-93c8-9cd491c6e29c">

<img width="537" alt="Screenshot 2024-05-03 at 10 17 11" src="https://github.com/wudsn/wudsn-ide-install/assets/7768922/b748f221-53c8-48d2-87c0-20880c486931">

**Comment by Sasatari** [comment]
> Here's a workaround that codesigns the Eclipse.app folder, but that can break at any time as Eclipse writes files to it.
> 
> [wudsn.command.zip](https://github.com/wudsn/wudsn-ide-install/files/15180773/wudsn.command.zip)

Just setting up my new MBP M4 Max and had the same issue, but with this new wudsn.command all works like expected. Should be communicated better, if I wouldn´t have the idea to check the Issues here, I would have been lost.

Thanks for solving it. :)

**Comment by peterdell** [comment]
> Just setting up my new MBP M4 Max and had the same issue, but with this new wudsn.command all works like expected. Should be communicated better, if I wouldn´t have the idea to check the Issues here, I would have been lost.
> Thanks for solving it. :)

I was waiting for feedback. I have included the forced signing now in the standard scripts and downloads. with 235af61cf5ce6c9998b34e95b05d1687921eb591.  I've also tested it successfully on my M1.

**Comment by peterdell** [comment]
> The only feature not working correctly is code hinting/completition.

I have moved this to a separate issue for the IDE at https://github.com/wudsn/wudsn-ide/issues/25 because it is not an installer issue. The answer is also provided there.
